contrib/platform/src/com/sun/jna/platform/win32/Kernel32Util.java

```diff
@@ -678,4 +682,149 @@ public static final String extractVolumeGUID(String volumeGUIDPath) {
         
         return volumeGUIDPath.substring(VOLUME_GUID_PATH_PREFIX.length(), volumeGUIDPath.length() - VOLUME_GUID_PATH_SUFFIX.length());
     }
+
+	/**
+	 * Gets the specified resource out of the specified executable file
+	 * 
+	 * @param path
+	 *            The path to the executable file
+	 * @param type
+	 *            The type of the resource (either a type name or type ID is
+	 *            allowed)
+	 * @param name
+	 *            The name or ID of the resource
+	 * @return The resource bytes, or null if no such resource exists.
+	 */
+	public static byte[] getResource(String path, String type, String name) {
+		Kernel32 kernel32 = Kernel32.INSTANCE;
```

Comment:
Code inconsistency - here you use a local variable that holds _Kernel32.INSTANCE_ and further down (in _getResourceNames_) you are not doing that. Please adhere to the same style as the other code in this class - i.e., user Kernel32.INSTANCE explicitly every time you need to access it.